optaplanner-examples/src/main/java/org/optaplanner/examples/meetingscheduling/optional/score/MeetingSchedulingConstraintProvider.java

```diff
@@ -0,0 +1,224 @@
+                .join(MeetingAssignment.class,
+                        filtering((leftAttendance, rightAttendance,
+                                leftAssignment) -> leftAttendance.getMeeting() == leftAssignment.getMeeting()),
+                        filtering((leftAttendance, rightAttendance,
+                                leftAssignment) -> leftAssignment.getStartingTimeGrain() != null))
+                .join(MeetingAssignment.class,
+                        filtering((leftAttendance, rightAttendance, leftAssignment,
+                                rightAssignment) -> rightAttendance.getMeeting() == rightAssignment.getMeeting()),
+                        filtering((leftAttendance, rightAttendance, leftAssignment,
+                                rightAssignment) -> rightAssignment.calculateOverlap(leftAssignment) > 0))
+                .penalizeConfigurable("Preferred attendance conflict");
+    }
+
+    // ************************************************************************
+    // Soft constraints
+    // ************************************************************************
+
+    protected Constraint doMeetingsAsSoonAsPossible(ConstraintFactory constraintFactory) {
+        return constraintFactory.fromUnfiltered(MeetingAssignment.class)
+                .filter(meetingAssignment -> meetingAssignment.getStartingTimeGrain() != null)
+                .penalizeConfigurable("Do all meetings as soon as possible",
+                        MeetingAssignment::getLastTimeGrainIndex);
+    }
+
+    protected Constraint oneBreakBetweenConsecutiveMeetings(ConstraintFactory constraintFactory) {
+        return constraintFactory.fromUnfiltered(MeetingAssignment.class)
+                .filter(meetingAssignment -> meetingAssignment.getStartingTimeGrain() != null)
+                .join(MeetingAssignment.class,
+                        filtering((leftAssignment, rightAssignment) -> rightAssignment.getStartingTimeGrain() != null),
+                        filtering((leftAssignment,
+                                rightAssignment) -> leftAssignment
+                                        .getLastTimeGrainIndex() == rightAssignment.getStartingTimeGrain().getGrainIndex() - 1))
+                .penalizeConfigurable("One TimeGrain break between two consecutive meetings");
+    }
+
+    protected Constraint overlappingMeetings(ConstraintFactory constraintFactory) {
+        return constraintFactory.fromUnfiltered(MeetingAssignment.class)
+                .filter(meetingAssignment -> meetingAssignment.getStartingTimeGrain() != null)
+                .join(MeetingAssignment.class,
+                        filtering((leftAssignment, rightAssignment) -> rightAssignment.getStartingTimeGrain() != null),
+                        filtering((leftAssignment,
+                                rightAssignment) -> rightAssignment.getMeeting().getId() < leftAssignment.getMeeting().getId()),
```

Comment:
`lessThan()`?

Reply:
Similar to `equal()`, how to put lambda expression in `lessThan()`?

Reply:
You need to move past method references. Simply use full lambda syntax - `(a, b) -> c`. That will allow you to write all joiners without restrictions.